DuckDuckGo/Updates/UpdateController.swift

```diff
@@ -151,80 +142,68 @@ final class UpdateController: NSObject, UpdateControllerProtocol {
         }
     }
 
-    func checkForUpdate() {
+    func checkForUpdateIfNeeded() {
+        guard let updater, !updater.sessionInProgress else { return }
+
         Logger.updates.debug("Checking for updates")
```

Comment:
There was a mistake when refactoring pixels and all update pixels are just debug pixels. Do you think it would be beneficial to make them available in release builds for potential debugging?

Reply:
It may be beneficial to have pixels at different states to see if our users make it to the end of the update cycle. Have we defined those, or do they have to be created?

Reply:
There are couple of pixels and the error pixel in case something goes wrong. Monitoring various states as you suggested may be a good idea too 👍